firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Driver/Serial/Backend/UART.h

```diff
@@ -10,17 +10,17 @@
 #include "Pufferfish/Application/States.h"
 #include "Pufferfish/Driver/Serial/Backend/Backend.h"
 #include "Pufferfish/HAL/Interfaces/CRCChecker.h"
-#include "Pufferfish/HAL/STM32/HALBufferedUART.h"
+#include "Pufferfish/HAL/STM32/BufferedUART.h"
 
 namespace Pufferfish::Driver::Serial::Backend {
 
 class UARTBackend {
  public:
-  using BufferedUART = HAL::LargeBufferedUART;
+  using BufferedUART = HAL::STM32::LargeBufferedUART;
```

Comment:
It looks like this renaming has exposed another problem with dependencies: `Driver/Serial/Backend/UART.h` should only be relying on a `HAL::Interfaces::BufferedUART`, not a `HAL::STM32::LargeBufferedUART`. Can you try to fix this problem in this PR?